src/openxr/OpenXRApi.cpp

```diff
@@ -2546,7 +2568,13 @@ bool OpenXRApi::poll_events() {
 				// TODO: do something
 			} break;
 			default:
-				Godot::print_warning(String("OpenXR Unhandled event type ") + String::num_int64(runtimeEvent.type), __FUNCTION__, __FILE__, __LINE__);
+				bool handled = false;
+				for (XRExtensionWrapper *wrapper : registered_extension_wrappers) {
```

Comment:
I would place this above our own switch so the extensions can also react to events that we're already reacting too. 

So it becomes:
```
bool handled = false;
for (XRExtensionWrapper *wrapper : registered_extension_wrappers) {
	handled |= wrapper->on_event_polled(runtimeEvent);
}
switch (runtimeEvent.type) {
	...
	default: {
		if (!handled) {
			Godot::print_warning(String("OpenXR Unhandled event type ") + String::num_int64(runtimeEvent.type), __FUNCTION__, __FILE__, __LINE__);
		}
 	} break;
 }
```
All though admittedly it may be a little double seeing we call into the wrapper for some of the events already.

Also small nitpick but it's up to you if you want to follow suit, I always add a code block in each case of a switch including (so `default: { ... } break;`). It's a style of programming where you assume people who come after you may not understand your code fully and make a change earlier on in code that break because everything is in the same scope. It also enforces a visual consistency making anything that breaks the pattern stand out. 
I do the same with `if` statements, you'll rarely see me do one lines instead always writing out the if statement block fully as `if (condition) { code }`

Again it's really something I just adopted a long time ago so I don't expect anyone to follow suit, but seeing it breaks the pattern if someone doesn't it always stands out to me :)

Reply:
That's a good call actually. I've myself ran into this issue several times.